pkg/controller/elasticsearch/user/filerealm/role_users.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package filerealm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/cloud-on-k8s/pkg/utils/set"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/stringsutil"
+)
+
+// roleUsersMapping is a map {role name -> [] user names}
+type roleUsersMapping map[string][]string
+
+// mergeWith merges multiple usersPasswordHashes, giving priority to other.
+func (r roleUsersMapping) mergeWith(other roleUsersMapping) roleUsersMapping {
+	if len(r) == 0 {
+		return other
+	}
+	if len(other) == 0 {
+		return r
+	}
+	for otherRole, otherUsers := range other {
+		currentUsers, exists := r[otherRole]
+		if !exists {
+			// role does not exist yet, create it
+			r[otherRole] = otherUsers
```

Comment:
Should we sort here too?

Reply:
I would expect `other` to be already made of sorted users.
Whereas we need to sort when merging the 2 users slices.

Reply:
see https://github.com/elastic/cloud-on-k8s/pull/2682#discussion_r390846007